File: app/src/main/java/com/commit451/nativestackblur/sample/BlurTransformation.java
package com.commit451.nativestackblur.sample;

import android.graphics.Bitmap;

import com.commit451.nativestackblur.NativeStackBlur;
import com.squareup.picasso.Transformation;

/**
 * Created by John on 9/10/15.
 */
public class BlurTransformation implements Transformation {

    private int mBlurRadius;

    public BlurTransformation(int blurRadius) {
        mBlurRadius = blurRadius;
    }

    @Override
    public Bitmap transform(Bitmap source) {
        Bitmap bm = NativeStackBlur.process(source, mBlurRadius);
        source.recycle();
        return bm;
    }

    @Override
    public String key() {
        return getClass().getCanonicalName() + "-" + mBlurRadius;
    }
}

File: app/src/main/java/com/commit451/nativestackblur/sample/BlurTransformation.kt
package com.commit451.nativestackblur.sample

import android.graphics.Bitmap
import coil.size.Size
import coil.transform.Transformation
import com.commit451.nativestackblur.NativeStackBlur

class BlurTransformation @JvmOverloads constructor(
    private val radius: Int = DEFAULT_RADIUS
) : Transformation {

    @Suppress("NullableToStringCall")
    override val cacheKey = "${BlurTransformation::class.java.name}-$radius"

    override suspend fun transform(input: Bitmap, size: Size): Bitmap {
        return NativeStackBlur.process(input, radius)
    }

    private companion object {
        private const val DEFAULT_RADIUS = 10
    }
}